### PR TITLE
CI/CD: update the version of go build

### PR DESCRIPTION
**Pull request by leyao-daily**
This PR is to update the go verison for rule server to 1.21 and controller-gen to 0.13.0

### CODE REVIEW

**Review by jcchavezs — APPROVED** [review]

File: .github/workflows/build-ruleserver.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Setup go
         uses: actions/setup-go@v1
         with:
-          go-version: "1.17"
+          go-version: "1.19"
```
Comment (jcchavezs) [inline .github/workflows/build-ruleserver.yaml:28]: wonder why not 1.21

Comment (leyao-daily) [reply .github/workflows/build-ruleserver.yaml]: fixed.

File: .github/workflows/build-ruleserver.yaml [hunk 2]
```diff
@@ -39,6 +39,12 @@ jobs:
       - name: Build server
         run: CGO_ENABLED=0 go build main.go
 
+      - name: Fix
```
Comment (jcchavezs) [inline .github/workflows/build-ruleserver.yaml:42]: Could we have more descriptive name for this and more important a comment on why this is needed?

File: .github/workflows/build-ruleserver.yaml [hunk 3]
```diff
@@ -39,6 +39,12 @@ jobs:
...
+        run: make controller-gen
+
+      - name: find
```
Comment (jcchavezs) [inline .github/workflows/build-ruleserver.yaml:45]: same as above

File: .github/workflows/build-ruleserver.yaml (hunk 2 above)
Comment (leyao-daily) [reply .github/workflows/build-ruleserver.yaml]: Hi, is just for testing to get the log, updating the go version brings many unexpected issues, and I meet some issues testing locally, so just print some log in CI/CD. Now it should work with updating go to 1.31 and controller-gen to 0.13.0, get -> install

**Comment by leyao-daily** [comment]
@jcchavezs any comments on it? :)

**Review by jcchavezs — APPROVED** [review]

**Pull request merged**